src/videoTrackingHTML5.js

```diff
@@ -0,0 +1,248 @@
+            eventTimestamp: e.target.currentTime,
+            currentVolume: e.target.volume
+        };
+        if (e.target.currentTime < e.target.duration && isUserSeeking == false) {
+            if (e.target.hasPaused == true){
+                eventData.eventType = "Seeked when paused";
+                trackEvent(eventData);
+            } else if (e.target.hasPlayed == false && e.target.hasPaused == false) {
+                eventData.eventType = "Seeked before playback";
+                trackEvent(eventData);
+            } else if (e.target.hasPlayed == true && e.target.hasPaused == false && e.target.hasReplayed == false && e.target.hasEnded == true) {
+                eventData.eventType = "Seeked after finished watching";
+                trackEvent(eventData);
+                //workaround for the fact that after resume type of "Play", "seeking during playback" is automatically triggered
+            } else if (e.target.hasPlayed == true && e.target.hasPaused == false && e.target.hasReplayed == false) {
+                eventData.eventType = "Seeked during playback";
+                trackEvent(eventData);
+            } 
+             else if (e.target.hasReplayed == true){
+                e.target.hasReplayed = false;
+            }
+        }
+    };
+
+    const processEndedMediaEvent = (e) => {
+        e.target.hasPaused = false;
+        e.target.hasEnded = true;
+        var eventData = {
+            eventType: "Watched",
+            videoTitle: getVideoName(e.target),
+            eventTimestamp: e.target.currentTime,
+            currentVolume: e.target.volume
+        };
+        trackEvent(eventData);
+    };
+
+    const handleTabUnloadDuringVideoPlay = (e) => {
+        mediaElements = document.querySelectorAll("video");
+        mediaElements.forEach((mediaElement) => {
+            //4Patryk: I didn't have better idea for getting around eslint, it has issue with me checking for my own hasPlayed property on an object (it's added at the beginning of code execution)
+            // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+            // @ts-ignore
+            if (mediaElement.ended == false && mediaElement.paused == false && mediaElement.hasPlayed == true){
+                var currentPlayTime = mediaElement.currentTime;
+                var videoTitle = getVideoName(mediaElement);
+                var eventData = {
+                    eventType: "Tab unloaded during video play",
+                    videoTitle: videoTitle,
+                    eventTimestamp: currentPlayTime
+                };
+                trackEvent(eventData);
+            }
+    
+        });
+    };
+
+    const processVolumeChangeMediaEvent = (e) => {
+        let eventTypeToTrack;
+        let currentVolume = parseInt((e.target.volume*100).toFixed(0));
+        if (e.target.muted == true){
+            eventTypeToTrack = "Muted";
+            e.target.hasMuted = true;
+            lastTrackedVolume = 0;
```

Comment:
As I can see this variable is not used at all 🤔

Reply:
which one you mean? all of them are used from what I see

Reply:
I mean`lastTrackedVolume`. It is assigned it few places but not used in my opinion.